### PR TITLE
FOUR-16819 Implement rate limit for security log of request errors

### PR DESCRIPTION
**Pull request by caleeli**
## Solution
- Implement rate limit for security log of request errors

## How to Test
- Create a process with an script that throws an exception
- Run the process
- Security log should register only one RequestError per day per request

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-16819

## Code Review Checklist
- [x] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [x] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [x] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [x] This solution fixes the bug reported in the original ticket.
- [x] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [x] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [x] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [x] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [x] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:next

### CODE REVIEW

File: ProcessMaker/Events/RequestError.php [hunk 1]
```diff
@@ -57,4 +58,26 @@ public function getEventName(): string
...
         return 'RequestError';
     }
+
+    /**
+     * Dispatch the event if the process request is not rate limited
+     */
+    public static function dispatchIfNotRateLimited(ProcessRequest $request, string $error)
+    {
+        $key = 'process-request-errors:' . $request->getKey();
+        $limit= config('app.process_request_errors_rate_limit', 1);
```
Comment (julceslauhub) [inline ProcessMaker/Events/RequestError.php:68]: code style: add an empty space before the equal sign

File: ProcessMaker/Events/RequestError.php [hunk 2]
```diff
@@ -57,4 +58,26 @@ public function getEventName(): string
     {
         return 'RequestError';
     }
+
+    /**
+     * Dispatch the event if the process request is not rate limited
+     */
+    public static function dispatchIfNotRateLimited(ProcessRequest $request, string $error)
+    {
```
Comment (devmiguelangel) [inline ProcessMaker/Events/RequestError.php:66]: Add return type
```suggestion
    public static function dispatchIfNotRateLimited(ProcessRequest $request, string $error): bool
    {
```

File: config/app.php [hunk 3]
```diff
@@ -234,4 +234,8 @@
     'task_drafts_enabled' => env('TASK_DRAFTS_ENABLED', true),
 
     'force_https' => env('FORCE_HTTPS', true),
+
+    // Process Request security log rate limit: 1 per day (86400 seconds)
+    'process_request_errors_rate_limit' => env('PROCESS_REQUEST_ERRORS_RATE_LIMIT', 1),
```
Comment (devmiguelangel) [inline config/app.php:239]: Add `PROCESS_REQUEST_ERRORS_RATE_LIMIT` to .env.example file

File: config/app.php [hunk 4]
```diff
@@ -234,4 +234,8 @@
...
+    'process_request_errors_rate_limit_duration' => env('PROCESS_REQUEST_ERRORS_RATE_LIMIT_DURATION', 86400),
```
Comment (devmiguelangel) [inline config/app.php:240]: Add `PROCESS_REQUEST_ERRORS_RATE_LIMIT_DURATION` to .env.example file

**Review by devmiguelangel — CHANGES_REQUESTED** [review]

**Review by devmiguelangel — APPROVED** [review]

**Review by julceslauhub — APPROVED** [review]
Approved

**Pull request merged**